### PR TITLE
Added Authentication Functionality

### PR DESCRIPTION
**Pull request by munjalSaksham71**
1. Added Login/Signup Forms 
Note:  As I haven't created any homepage till now - so it won't redirect. [To verify: please refer to application tab in inspect element]
2. Created Header.

### CODE REVIEW

File: src/actions/authActions.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import axios from "axios";
+
+const config = {
+  headers: {
+    "content-type": "application/json",
+  },
+};
```
Comment (JantuDeb) [inline src/actions/authActions.js:7]: It's might not require here

**Review by JantuDeb — APPROVED** [review]
Overall it is looking fine

File: src/actions/authActions.js (hunk 1 above)
Comment (munjalSaksham71) [reply src/actions/authActions.js:7]: Yes, you are right -  i have added to make sure that my content in post request is in json always.. that the reason i have added.

File: src/Screens/SignupScreen/SignupScreen.jsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    const [password, setPassword] = useState('');
+    const [firstName, setFirstName] = useState('');
+    const [lastName, setLastName] = useState('');
+    const { user, signupUser } = useAuth();
+    const navigate = useNavigate();
+
+    const submitHandler = async (e) => {
+        e.preventDefault();
+        if(!username || !password || !firstName || !lastName ){
+            alert("Please enter All the fields");
```
Comment (paarthbajaj) [inline src/Screens/SignupScreen/SignupScreen.jsx:17]: Great alerts handling.

**Review by paarthbajaj — APPROVED** [review]
Good that you have used utility classes. I also appreciate your way of writing the code like handling all the things upfront like alerts, responses from API, etc. Well-written code brother. You can merge.

**Pull request merged**